api/v1/nifidataflow_types.go

```diff
@@ -165,11 +167,14 @@ func (d *NifiDataflowSpec) SyncNever() bool {
 	return d.GetSyncMode() == SyncNever
 }
 
-func (d *NifiDataflowSpec) GetParentProcessGroupID(rootProcessGroupId string) string {
-	if d.ParentProcessGroupID == "" {
+func (d *NifiDataflowSpec) GetParentProcessGroupID(rootProcessGroupId string, parentProcessGroup *ParentProcessGroup) string {
+	if parentProcessGroup != nil && parentProcessGroup.Type == ResourceProcessGroup && parentProcessGroup.UUID != "" {
+		return parentProcessGroup.UUID
+	} else if d.ParentProcessGroupID == "" {
 		return rootProcessGroupId
+	} else {
+		return d.ParentProcessGroupID
 	}
```

Comment:
```suggestion
	}
	return d.ParentProcessGroupID
```

`else` is not needed.